### ISSUE TITLE
Time Format Gives Error

### ISSUE THREAD

**Issue by cianphillips** [BrightLedSigns/LedSign#5]
The following script produces an "[INVALID TIME TAG]" error on the Model: m500n-7x80rg2 LED display after successfully uploading. Am I using the time 'markup' correctly? I tried to find the problem and I found the following a little confusion, though I don't know enough to call it a bug.

lib/LedSign/M500.pm at about line 697 looks like it's going to setup the time map but then uses the LedSign::M500::PAUSEMAP()->{$time} instead of what I thought would be LedSign::M500::TDMAP()->{$time}

#!/usr/bin/perl
    use LedSign::M500;	
	my $buffer="";
	my $sign="";
	my $sign=LedSign::M500->new();
	$sign->sendQueue(device => "/dev/tty.usbserial");
    $sign->sendCmd(
    	device => "/dev/tty.usbserial",
        setting => "settime",
        value => "now"
    );
    my $buffer=LedSign::M500->new();
    $buffer->queueMsg( data =>"`<t:A><c:RED>RED`");
    $buffer->sendQueue(
        device => '/dev/tty.usbserial'
    )

**Comment by budgetneon** [comment]
Correct on both counts.   It's a bug, and those two references around line 702 should be to TDMAP.   I pushed a change out just now.

**Comment by mik3rgit** [comment]
The time keys are also incorrect the documentation in file:
LedSign/LedSign::M500.md

The example shows <t:A> when it should be <t:DATE1>. You need to use the following keys:
DATE1, DATE2, DATE3, DATE4, TIME1, TIME2, TIME3

**Comment by budgetneon** [comment]
Okay...you're saying this is a documentation bug, correct, specifically with the M500 piece?

**Comment by mik3rgit** [comment]
Yes it's a documentation bug on the M500 piece. If you use <t:DATE1> then the time is correctly displayed on the device.

**Comment by budgetneon** [comment]
Pushed out a fix.  I didn't, though, document what the various date/time tags display.  I suspect what was described wasn't entirely correct, so you would have to experiment to see what they actually display.

**Comment by mik3rgit** [comment]
According to the cheat-sheet here:
http://wls.wwco.com/ledsigns/m-500/m-500-protocol.php

You have "DATE4" in the TDMAP, which is invalid. It causes the device to display "[INVALID TIME TAG]".  The code "^a" doesn't map to anything on the cheatsheet.

I'm not sure who owns this page, but they incorrectly reference "0x0C" instead of "0x0D" which is the carriage return.

**Comment by budgetneon** [comment]
I'm the person that made that cheat sheet and sent to that person :)

Again, there's no definitive source of truth for M500 signs, anyone that has anything has it because they hooked up an analyzer and tried things.    These signs have been around a while, and they have changed here and there...newer ones do a few things older ones don't.

Based on your other experience, I'm also not 100% convinced you have an M500.

So, in short, I'll hold off on further changes until I have some time to try it on a known good M500.
